### ISSUE TITLE
Feature: Display workspace parameters in the workspace page

### ISSUE THREAD

**Issue by BrunoQuaresma** [coder/coder#11837]
Having the workspace parameters on the workspace page can be helpful in debugging and understanding certain workspace behaviors. This was suggested by @mtojek .

**Comment by matifali** [comment]
A huge fan of this. Great.

**Comment by BrunoQuaresma** [comment]
I would say all the values from the workspace build parameters endpoint 🤔

**Comment by matifali** [comment]
@BrunoQuaresma IMO, that's extra. We should only show what the user has selected for his workspace. for example, the chosen region, etc.

**Comment by BrunoQuaresma** [comment]
I mean, if we can get this info from the API I'm ok with that but Idk if we have a way to filter by "user entered values"

**Comment by bpmct** [comment]
I don't think we need to show any of them as resource_metadata can be used to show the relevant parameters with proper formatting

**Comment by mtojek** [comment]
I have a feeling that we don't have great user experience right now. Consider my workspaces:

<img width="913" alt="Screenshot 2024-01-29 at 14 59 57" src="https://github.com/coder/coder/assets/14044910/239215bc-f6d6-4a4a-9bf8-377e10b912fa">
<img width="903" alt="Screenshot 2024-01-29 at 14 59 47" src="https://github.com/coder/coder/assets/14044910/7a2b2f5a-8d76-44d8-b2d3-0fd66353be13">

These are 2 different workspaces stopped and pending deletion. Unfortunately, Coder does not expose any parameters or workspace details here, so I don't know what is the workspace region, how many cores, how much memory, etc. + some parameter values to distinguish. 

@BrunoQuaresma You're a UX magician :) Maybe you could prepare some mockups we could A/B compare and decide how we want to continue?

**Comment by matifali** [comment]
Yes, Exposing the selected value of coder_parameters should really help improve UX. This can be done with `coder_metadata` too but that requires template editing. @bpmct I strongly recommend we render the `coder_parameter` values instead of relying on a template admin to add `coder_metadata.`

**Comment by bpmct** [comment]
If we put them as a sidebar item, I think that would be perfect. Our customers already use `resource_metadata` to display important values and parameters in their templates. Not all parameters (e.g. dotfiles URI) deserve constant real-estate on the workspace page.

**Comment by matifali** [comment]
I think we should still do this.
